yarn-project/archiver/src/archiver/data_retrieval.ts

```diff
@@ -139,11 +140,53 @@ async function getBlockFromRollupTx(
   if (!allowedMethods.includes(functionName)) {
     throw new Error(`Unexpected method called ${functionName}`);
   }
-  const [headerHex, archiveRootHex, , , , bodyHex] = args! as readonly [Hex, Hex, Hex, Hex[], ViemSignature[], Hex];
+  // TODO(#9101): 'bodyHex' will be removed from below
+  const [headerHex, archiveRootHex, , , , bodyHex, blobInputs] = args! as readonly [
+    Hex,
+    Hex,
+    Hex,
+    Hex[],
+    ViemSignature[],
+    Hex,
+    Hex,
+  ];
 
   const header = Header.fromBuffer(Buffer.from(hexToBytes(headerHex)));
+  // TODO(#9101): Retreiving the block body from calldata is a temporary soln before we have
+  // either a beacon chain client or link to some blob store. Web2 is ok because we will
+  // verify the block body vs the blob as below.
   const blockBody = Body.fromBuffer(Buffer.from(hexToBytes(bodyHex)));
 
+  const blockFields = blockBody.toFields();
+  // TODO(#9101): The below reconstruction is currently redundant, but once we extract blobs will be the way to construct blocks.
+  // The blob source will give us blockFields, and we must construct the body from them:
+  // TODO(#8954): When logs are refactored into fields, we won't need to inject them here.
+  const reconstructedBlock = Body.fromFields(
+    blockFields,
+    blockBody.noteEncryptedLogs,
+    blockBody.encryptedLogs,
+    blockBody.unencryptedLogs,
+  );
+
+  blockBody.squashEmptyLogs();
+  if (!reconstructedBlock.toBuffer().equals(blockBody.toBuffer())) {
+    // Sometimes the below will fail because empty logs can look like:
+    //   UnencryptedTxL2Logs { functionLogs: [ UnencryptedFunctionL2Logs { logs: [] } ]} OR UnencryptedTxL2Logs { functionLogs: []}
+    // Even though they represent the same data, hence squashing the blockBody logs above
```

Comment:
True - I added this for empty logs specifically so we wouldn't have to re-add empty structs in the same format (sometimes they have .functionLogs, sometimes not), and add an ugly clause to assign if empty, and gather from blob otherwise. I will take a different approach and remove the `squash` anyway!